standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/tools/MetaToolObjectStore.java

```diff
@@ -0,0 +1,841 @@
+
+  private void feedColumnSummary(MetadataTableSummary summary, ResultSet rs) throws SQLException {
+    String partitionColumn = rs.getString("PARTITION_COLUMN");
+    int partitionColumnCount = (partitionColumn == null) ? 0 : (partitionColumn.split(",")).length;
+    summary
+        .partitionSummary(partitionColumnCount, rs.getInt("PARTITION_CNT"))
+        .columnSummary(
+            rs.getInt("TOTAL_COLUMN_COUNT"),
+            rs.getInt("ARRAY_COLUMN_COUNT"),
+            rs.getInt("STRUCT_COLUMN_COUNT"),
+            rs.getInt("MAP_COLUMN_COUNT"));
+  }
+
+  private void feedTabFormatSummary(MetadataTableSummary summary, ResultSet rs) throws SQLException {
+    String tblType = rs.getString("TBL_TYPE");
+    String fileType = extractFileFormat(rs.getString("SLIB"));
+    Set<String> nonNativeTabTypes = new HashSet<>(Arrays.asList("jdbc", "kudu", "hbase", "iceberg"));
+    if (nonNativeTabTypes.contains(fileType)) {
+      tblType = fileType.toUpperCase();
+      if ("iceberg".equalsIgnoreCase(tblType)) {
+        String writeFormatDefault = rs.getString("WRITE_FORMAT_DEFAULT");
+        if (writeFormatDefault == null) {
+          writeFormatDefault = "null";
+        }
+        if (writeFormatDefault.equals("null")) {
+          fileType = "parquet";
+        } else {
+          fileType = writeFormatDefault;
+        }
+      }
+    } else if (TableType.MANAGED_TABLE.name().equalsIgnoreCase(tblType)) {
+      String transactionalProperties = rs.getString("TRANSACTIONAL_PROPERTIES");
+      if (transactionalProperties == null) {
+        transactionalProperties = "null";
+      }
+      tblType = "insert_only".equalsIgnoreCase(transactionalProperties.trim()) ?
+          "HIVE_ACID_INSERT_ONLY" : "HIVE_ACID_FULL";
+    } else if (TableType.EXTERNAL_TABLE.name().equalsIgnoreCase(tblType)) {
+      tblType = "HIVE_EXTERNAL";
+    } else {
+      tblType = tblType != null ? tblType.toUpperCase() : "NULL";
+    }
+    String compressionType = rs.getString("IS_COMPRESSED");
+    if (compressionType.equals("0") || compressionType.equals("f")) {
+      compressionType = "None";
+    }
+    summary.tableFormatSummary(tblType, compressionType, fileType);
+  }
+
+  private boolean prepareGetMetaSummary(JDOConnection jdoConn) throws MetaException {
+    try {
+      List<String> prepareQueries = sqlGenerator.getCreateQueriesForMetastoreSummary();
+      if (prepareQueries == null || prepareQueries.isEmpty()) {
+        LOG.warn("Metadata summary has not been implemented for dbtype {}", DatabaseProduct.dbType);
+        return false;
+      }
+      try (Statement stmt = ((Connection) jdoConn.getNativeConnection()).createStatement()) {
+        long startTime = System.currentTimeMillis();
+        for (String q : prepareQueries) {
+          stmt.addBatch(q);
+        }
+        stmt.executeBatch();
+        long endTime = System.currentTimeMillis();
+        LOG.info("Total query time for generating HMS Summary: {} (ms)", endTime - startTime);
```

Comment:
Should implement a Timeout for this tool? We have seen in production environments where this tool ran for 2 hrs and had to kill it manually.

Reply:
make sense, will add the timeout check